lib/fluent/plugin/out_dynatrace.rb

```diff
@@ -110,25 +117,46 @@ def user_agent
         "fluent-plugin-dynatrace v#{DynatraceOutputConstants.version}"
       end
 
-      def prepare_request(uri)
+      def prepare_request
+        log.on_trace { log.trace('#prepare_request') }
         req = Net::HTTP::Post.new(uri, { 'User-Agent' => user_agent })
         req['Content-Type'] = 'application/json; charset=utf-8'
         req['Authorization'] = "Api-Token #{@api_token}"
 
         req
       end
 
-      def send_to_dynatrace(body)
+      def synchronized_send_records(records)
+        log.on_trace { log.trace('#synchronized_send_records') }
         HTTP_REQUEST_LOCK.synchronize do
-          agent.start unless agent.started?
+          send_records(records)
+        end
+      end
 
-          req = prepare_request(@uri)
-          res = @agent.request(req, body)
+      def send_records(records)
+        log.on_trace { log.trace('#send_records') }
 
-          return if res.is_a?(Net::HTTPSuccess)
+        agent.start unless agent.started?
 
-          raise failure_message res
-        end
+        res = request(serialize(records))
+
+        return if res.is_a?(Net::HTTPSuccess)
+
+        raise failure_message res
```

Comment:
```suggestion
        response = send_request(serialize(records))

        return if response.is_a?(Net::HTTPSuccess)

        raise failure_message response
```
It's not that obvious that this already sends the request and returns the response.